tests/lambda.rs

```diff
@@ -90,8 +97,16 @@ impl Analysis<Lambda> for LambdaAnalysis {
 
     fn modify(egraph: &mut EGraph, id: Id) {
         if let Some(c) = egraph[id].data.constant.clone() {
-            let const_id = egraph.add(c);
-            egraph.union(id, const_id);
+            let added = egraph.add_instantiation(&c.1, &Default::default());
```

Comment:
Since unions are delayed, analyses must add the pattern to the egraph before doing the union in order to call `union_with_justification`. Otherwise, the sub-terms in the pattern may not be unioned with the child yet.